sdk/src/ed25519_instruction.rs

```diff
@@ -0,0 +1,287 @@
+#![cfg(feature = "full")]
+
+use crate::instruction::Instruction;
+use ed25519_dalek::{ed25519::signature::Signature, Signer, Verifier};
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Debug, PartialEq)]
+pub enum Ed25519Error {
+    InvalidPublicKey,
+    InvalidSignature,
+    InvalidDataOffsets,
+    InvalidInstructionDataSize,
+}
+
+pub const PUBKEY_SERIALIZED_SIZE: usize = 32;
+pub const SIGNATURE_SERIALIZED_SIZE: usize = 64;
+pub const SIGNATURE_OFFSETS_SERIALIZED_SIZE: usize = 11;
+pub const DATA_START: usize = SIGNATURE_OFFSETS_SERIALIZED_SIZE + 1;
+
+#[derive(Default, Serialize, Deserialize, Debug)]
+pub struct Ed25519SignatureOffsets {
+    signature_offset: u16,            // offset to ed25519 signature of 64 bytes
+    signature_instruction_index: u8,  // instruction index to find signature
```

Comment:
Know why the "instruction" in "signature_instruction_index"?  Couldn't it just be "signature_index"?